bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Runtime.java

```diff
@@ -72,40 +96,21 @@ public static Runtime getCurrentRuntime() {
       * @param metadata   Meta data of new strand.
       * @param callback   Callback which will get notify once method execution done.
       * @param args       Ballerina function arguments.
-      * @return the result of the function invocation
+      * @return           the result of the function invocation
+      * @deprecated       Use {@link #invokeMethodAsync(BObject, String, String, StrandMetadata, boolean,
+      *                   Callback, Map, Type, Object...)} providing isolation of method call.
       */
      public Object invokeMethodAsync(BObject object, String methodName, String strandName, StrandMetadata metadata,
                                      Callback callback, Object... args) {
          Function<?, ?> func = o -> object.call((Strand) (((Object[]) o)[0]), methodName, args);
```

Comment:
can remove most of this method's body by calling `invokeMethodAsync(BObject, String, String, StrandMetadata, boolean, Callback, Map, Type, Object...)` here.